tests/modules/text_embedding/test_embedding.py

```diff
@@ -788,3 +821,97 @@ def test_env_val_to_int():
+    ],
+)
+def test_sum_token_count_no_truncation(texts, expected_count, loaded_model):
+    tokenized, _ = loaded_model.model._truncate_input_tokens(
+        truncate_input_tokens=-1,  # don't truncate. Model's truncation can still apply.
+        texts=texts,
+    )
+    token_count = sum_token_count(
+        tokenized,
+        truncate_only=False,
+    )
+
+    assert token_count == expected_count
+
+
+@pytest.mark.parametrize(
+    ["texts", "truncate", "expected_count"],
+    [
+        # Only tokens requiring model attention is counted.
+        # [PAD] doesn't attract model attention, but [CLS] and [SEP] does
+        # [CLS] 5 normal [SEP]
+        (["12345"], 10, 7),
+        # [CLS] 3 normal [SEP] + [CLS] 2 normal [SEP] [PAD]
+        (["12345"], 5, 5 + 4),
+        # [CLS] 3 normal [SEP] + [CLS] 2 normal [SEP] [PAD], [CLS] 3 normal [SEP] + [CLS] 1 normal [SEP] [PAD] [PAD]
+        (["12 345", "6 789"], 5, 9 + 8),
+    ],
+)
+def test_sum_token_count_with_truncation(texts, truncate, expected_count, loaded_model):
+    tokenized, _ = loaded_model.model._truncate_input_tokens(
+        truncate_input_tokens=truncate,
+        texts=texts,
+    )
+    token_count = sum_token_count(
+        tokenized,
+        truncate_only=True,
+    )
+
+    assert token_count == expected_count
+
+
+def test_encoding_order(loaded_model: EmbeddingModule):
+    """Confirm that encoding doesn't modify the original sort order"""
+    separate_embeddings = [loaded_model.run_embedding(text=i) for i in MANY_INPUTS]
+    combined_embeddings = loaded_model.run_embeddings(texts=MANY_INPUTS)
+
+    separate_vectors = [
+        e.to_dict()["result"]["data"]["values"] for e in separate_embeddings
+    ]
+    combined_vectors = [
+        e["data"]["values"] for e in combined_embeddings.to_dict()["results"]["vectors"]
+    ]
+
+    assert len(separate_vectors) == len(
+        combined_vectors
+    ), "expected the same number separate and combined embeddings"
+
+    # test order by comparing value of individual embeddings in sequence
+    for i, e in enumerate(separate_vectors):
+        assert approx(e) == combined_vectors[i]
+
+    # test expected failure case by reordering
+    shifted_separate_vectors = separate_vectors[1:] + [separate_vectors[0]]
+
+    for i, e in enumerate(shifted_separate_vectors):
+        assert e != separate_vectors[i], "expected order to be have been altered"
+        assert (
+            not approx(e) == combined_vectors[i]
+        ), "expected altered order to not match combined vectors"
+
+
+@pytest.mark.parametrize(
+    ("mapping", "expected"),
+    [
+        ([0, 0, 0, 0, 0], [0]),
+        ([0, 1, 2, 3, 4], [0, 1, 2, 3, 4]),
+        ([0, 0, 1, 1, 1, 2], [0, 2, 5]),
+        ([], []),
+    ],
+)
+def test_get_sample_start_indexes(mapping, expected):
+    mock_tokenized = {"overflow_to_sample_mapping": mapping}
+    assert get_sample_start_indexes(mock_tokenized) == expected  # type: ignore
```

Comment:
q - are these not lists? (what is ignored here?)

Reply:
Thanks for this comment. This forced me to review and I found a bug!
I've updated the test inputs to be tensors. As a result I saw that `get_sample_start_indexes` also required a bug fix and the expected token count for `test_sum_token_count_with_truncation()` needed to be corrected. It previously counted all, instead of just the truncate component.

To answer your question:
- `mapping` is a is a `torch.Tensor` in a `BatchEncoding` object.
- It is difficult to directly or indirectly craft a `BatchEncoding` object precisely, hence why I opted to mock it. 
- `get_sample_start_indexes()` should return the index of the first occurance of each value in mapping:
`Tensor([0, 0, 1, 1, 1, 2])` -> `[0, 2, 5]` which represents the indexes in `Tensor([0, 0, 1, 1, 1, 2])` where the 1st occurrence of each number is located.

When input text is truncated by the tokenizer, it gets split into multiple encodings. `"overflow_to_sample_mapping"` holds the mapping of which encodings (index of mapping elements) relates to which input text (values of mapping elements).

Reply:
thanks for the explanation! is the `# type: ignore` comment portion now removable then?

Reply:
I'll remove the `#type: ignore`.
Linters will still complain since `get_sample_start_indexes` expects a `BatchEncoding` object while in the test it is provided a `dict` mock object with the minimum required keys to make `get_sample_start_indexes` work.